doc/lua_api.txt

```diff
@@ -3303,33 +3299,91 @@ The following functions provide escape sequences:
 
 Spatial Vectors
 ===============
-A spatial vector is similar to a position, but instead using
-absolute world coordinates, it uses *relative* coordinates, relative to
-no particular point.
-
-Internally, it is implemented as a table with the 3 fields
-`x`, `y` and `z`. Example: `{x = 0, y = 1, z = 0}`.
-However, one should *never* create a vector manually as above, such misbehavior
-is deprecated. The vector helpers set a metatable for the created vectors which
-allows indexing with numbers, calling functions directly on vectors and using
-operators (like `+`). Furthermore, the internal implementation might change in
+
+Minetest stores 3-dimensional spatial vectors in Lua as tables of 3 coordinates,
+and has a class to represent them (`vector.*`), which this chapter is about.
+For details on what a spatial vectors is, please refer to Wikipedia:
+https://en.wikipedia.org/wiki/Euclidean_vector.
+
+Spatial vectors are used for various things, including, but not limited to:
+
+* positions
+* position differences
+* velocities
+* accelerations
+* euler angles (Yes, using spatial vectors here has no real semantic meaning. It's
+  just 3 values. Therefore, most operations make no sense and should not be used
+  on vectors of euler angles.)
+* dimensions (Sizes in 3 directions.)
+
+Note that they are *not* used for:
+
+* n-dimensional vectors where n is not 3 (ie. n=2)
+* arrays of the form `{num, num, num}`
+
+The API documentation may refer to spatial vectors by any of the following notations:
+
+* `(x, y, z)` (Used rarely, and only if it's clear that it's a vector.)
+* `vector.new(x, y, z)`
+* `{x=num, y=num, z=num}`
+
+They are all (if not stated otherwise) referring to the same thing, that is: vectors,
+as produced by `vector.new`.
+
+Compatibility notes
+-------------------
+
+Internally, a vector is implemented using a table with the 3 fields `x`, `y` and `z`.
+It's similar to this: `{x = 0, y = 1, z = 0}`, but with a metatable.
+
+Historically (before Minetest 5.5.0), vectors used to be defined as tables of the
+form `{x = num, y = num, z = num}`, without any metatable. You might find this in
+old code.
+
+Most functions by Minetest that take vectors will continue to able to handle them,
```

Comment:
It is supposed to be vague.

Possible alternatives:

* "All functions that existed before 5.5.0": Then it's illegal to change an old function to become incompatible. Reviewers might have to check since when a function existed.
* "All functions": Might be annoying at some point in the far future if someone wants to add a new helper or something.
* "No functions [...] are guaranteed to": Would break most mods, on paper.

Coredevs can decide on such policies.

Reply:
Suggestion which keeps it vague and is thought to replace L3333 to L3340.
```
Manually constructed tables are deprecated and highly discouraged. This
interface should be used to ensure seamless compatibility between mods and the
Minetest API. This is especially important to callback function parameters and
functions overwritten by mods.
```
Side note: `to able to handle` sounds wrong. Typo?

Reply:
"Continue to"  "be able to"  "handle" sounds fine to a native English speaker and there's not really any more grammatically correct or fluent way to put it.

Reply:
Typo, the "be" was missing.
Applying suggestion.